### PR TITLE
Added user/team models

### PR DESCRIPTION
**Pull request by aniveera1**
- Fixed some Docker configs

### CODE REVIEW

File: backend/gamify/models.py [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
-# Create your models here.
+
+class Team(models.Model):
+    name = models.CharField(max_length=100, blank=False, null=False)
+    join_code = models.CharField(max_length=100, blank=False, null=False)
+
+
+class User(models.Model):
+    name = models.CharField(max_length=100, blank=False, null=False)
+    team = models.ForeignKey(Team, on_delete=models.CASCADE, blank=True, null=True)  # many-to-one
```
Comment (bestvibes) [inline backend/gamify/models.py]: We don't want to cascade here right? if a team gets deleted all the users of the team shouldn't get deleted

File: backend/gamify/models.py [hunk 2]
```diff
@@ -1,3 +1,12 @@
 from django.db import models
 
-# Create your models here.
+
+class Team(models.Model):
+    name = models.CharField(max_length=100, blank=False, null=False)
+    join_code = models.CharField(max_length=100, blank=False, null=False)
```
Comment (bestvibes) [inline backend/gamify/models.py:9]: Can we automatically generate the join code in the constructor for this?

File: backend/gamify/models.py [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+
+
+class User(models.Model):
```
Comment (bestvibes) [inline backend/gamify/models.py:17]: can we also have a constructor for this that defaults to just a name with no team

**Review by bestvibes — CHANGES_REQUESTED** [review]

**Comment by bestvibes** [comment]
oh also can you create serializers for these models please? that'll allow us to easily serialize to/from json for our requests/responses.

File: backend/README.md [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
-3) run `docker-compose up`
-4) exit using Ctrl-C or `docker-compose down`
+3) run `docker-compose up -d`
+4) `docker ps`, note container ID of backend_web image
+5) `docker exec -it [CONTAINER_ID] bash`
+7) `python manage.py migrate`
+8) exit using Ctrl-C or `docker-compose down`
+
+- This is super hacky because you have to run migrations manually
+- The alternative is to build out some logic where the docker web container waits for the docker db container to be ready
```
Comment (bestvibes) [inline backend/README.md]: Look at this Dockerfile and .docker/entrypoint.sh on how to handle this, i think it'll make our lives a lot easier
https://github.com/CS130-W20/team-B1/tree/master/backend

File: backend/gamify/models.py (hunk 3 above)
Comment (aniveera1) [reply backend/gamify/models.py:17]: Doesn't Django's built in constructor already do that b/c the other two fields are nullable?

**Comment by aniveera1** [comment]
We can use Django's built in serializers?

File: backend/gamify/models.py [hunk 5]
```diff
@@ -1,3 +1,18 @@
...
-# Create your models here.
+
+class Team(models.Model):
+    name = models.CharField(max_length=100, blank=False, null=False)
+    join_code = models.CharField(max_length=100, blank=False, null=False)
+
+    # Factory method b/c messing with constructor might break other things
+    def create(cls):
+        # src: https://stackoverflow.com/questions/16008670/how-to-hash-a-string-into-8-digits
+        return Team(name=name, join_code=hashlib.sha256(name.encode('utf-8')).hexdigest()[:5])
```
Comment (bestvibes) [inline backend/gamify/models.py]: wait doesn't this assume unique team names which we decided isn't enforced?

File: backend/gamify/models.py (hunk 3 above)
Comment (bestvibes) [reply backend/gamify/models.py:17]: oh yeah maybe, cool nvm

**Comment by bestvibes** [comment]
> We can use Django's built in serializers?

yeah, but we still need to subclass them once for each model in a `serializers.py` file.

**Pull request merged**